plugins/out_azure_kusto/azure_kusto_conf.c

```diff
@@ -462,31 +466,29 @@ int azure_kusto_load_ingestion_resources(struct flb_azure_kusto *ctx,
                             execute_ingest_csl_command(ctx, ".get kusto identity token");
 
                         if (response) {
+                            if (pthread_mutex_lock(&ctx->resources_mutex)) {
+                                flb_plg_error(ctx->ins, "error locking resources mutex");
+                                return -1;
+                            }
                             identity_token =
                                 parse_ingestion_identity_token(ctx, response);
 
                             if (identity_token) {
-                                ret = flb_azure_kusto_resources_clear(ctx->resources);
```

Comment:
Should we remove this teardown function?
The previous `blob_ha`, `queue_ha`, `identitiy_token` should be cleaned up.

Reply:
added back the tear down function